src/systems/scene_broadcaster/SceneBroadcaster.cc

```diff
@@ -294,6 +308,18 @@ void SceneBroadcaster::PostUpdate(const UpdateInfo &_info,
       this->dataPtr->stateCv.notify_all();
     }
 
+    // process async state requests
+    if (!this->dataPtr->stateRequests.empty())
+    {
+      ignition::msgs::SerializedStepMap res;
+      res.CopyFrom(this->dataPtr->stepMsg);
+      for (const auto &reqSrv : this->dataPtr->stateRequests)
+      {
+        this->dataPtr->node->Request(reqSrv, res);
```

Comment:
when the async request is received, I set the `this->stateServiceRequest` variable to `true` so that should make sure the full state is populated